### PR TITLE
xcodebuild action

### PR DESCRIPTION
**Pull request by dtrenz**
Exposes xcodebuild CLI tool via fastlane action with defaults, environment variables & other nice things.

Usage:

``` ruby
lane :build do
  # Create an archive. (./build-dir/MyApp.xcarchive)
  xcodebuild(
    archive: true,
    archive_path: './build-dir/MyApp.xcarchive',
    scheme: 'MyApp',
    workspace: 'MyApp.xcworkspace'
  )

  # Export a signed binary (./build-dir/MyApp.ipa)
  xcodebuild(
    export_archive: true,
    export_path: './build-dir/MyApp'
  )
end
```

The action supports the following .env vars:
- PROJECT
- WORKSPACE
- SCHEME
- BUILD_PATH

For example, if your .env file looked like this:

```
PROJECT="./MyApp.xcodeproj"
WORKSPACE="./MyApp.xcworkspace"
SCHEME="MyApp"
BUILD_PATH="./build"
```

...then you can simplify the lane usage to:

``` ruby
lane :build do
  # Create an archive. (./build/MyApp.xcarchive)
  xcodebuild(
    archive: true
  )

  # Export a signed binary (./build/MyApp.ipa)
  xcodebuild(
    export_archive: true
  )
end
```

When archiving, the path to the .xcarchive file is cached in `SharedValues` for xcodebuild to use in a subsequent export.

When exporting, the path to the generated ipa is cached in `SharedValues::IPA_OUTPUT_PATH` for any deploy steps (i.e. Crashlytics) to use (should this also store in `ENV["IPA_OUTPUT_PATH"]` as the `ipa` action does now?).

I'd like to add even simpler alias actions for the most common xcodebuild operations; archive, export, build, & test.  Something like: 'xcarchive', 'xcbuild', 'xcexport', and 'xctest'.

This could reduce the usage to a single action verb, for simpler projects.

I was thinking of bundling these into the same module/file as alias classes. Thoughts?

Example usage of alias actions in a complete continuous deployment lane:

``` ruby
lane :inhouse do
  # Run tests
  xctest(
    destination: "name=iPhone 5s,OS=8.1"
  )

  # Generate XML code coverage
  gcovr(
    xml: true,
    output: "./code-coverage/report.xml"
  )

  # Clean the project
  xcclean

  # Create an archive
  xcarchive

  # Fetch/install provisioning profile
  sigh

  # Export a signed IPA
  xcexport

  # Deploy build to Beta
  crashlytics

  # Notify the HipChat room that the build was released
  hipchat(
    message: "iOS build successfully deployed to Beta!",
    channel: ENV["HIPCHAT_ROOM"],
    success: true
  )
end
```

### CODE REVIEW

File: fastlane.gemspec [hunk 1]
```diff
@@ -26,9 +26,10 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'slack-notifier', '~> 1.0' # Slack notifications
   spec.add_dependency 'shenzhen', '>= 0.12.1' # to upload to Hockey and Crashlytics
```
Comment (joshdholtz) [inline fastlane.gemspec]: @dtrenz Did you still need to remove this dependency?

**Comment by joshdholtz** [comment]
> I'd like to add even simpler alias actions for the most common xcodebuild operations; archive, export, build, & test. Something like: 'xcarchive', 'xcbuild', 'xcexport', and 'xctest'.
>
> I was thinking of bundling these into the same module/file as alias classes. Thoughts?

^^ Yes yes, I agree with this!

File: lib/fastlane/actions/xcodebuild.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        end
+
+        # The args we will build with
+        cli_args = Array[]
+
+        # Supported ENV vars
+        build_path = ENV["BUILD_PATH"]
+        scheme     = ENV["SCHEME"]
+        workspace  = ENV["WORKSPACE"]
+        project    = ENV["PROJECT"]
```
Comment (joshdholtz) [inline lib/fastlane/actions/xcodebuild.rb]: @dtrenz Are there ENV variables required? I see `build_path` and `scheme` being used below (https://github.com/KrauseFx/fastlane/pull/104/files#diff-6068f34d3ade6167ee700963b4ace891R78) but only possibly being instantiated here

File: fastlane.gemspec (hunk 1 above)
Comment (dtrenz) [inline fastlane.gemspec]: Yes, but I wasn't sure if I should do it, or wait for @KrauseFx to fix it and then merge in the fix.

**Comment by dtrenz** [comment]
Added alias/helper/companion actions:

**xcarchive**
```ruby
xcarchive(
  archive_path: "./build-dir/MyApp.xcarchive",
  scheme: "MyApp",
  workspace: "MyApp.xcworkspace"
)
```
**xcbuild**
```ruby
xcbuild(
  scheme: "MyApp",
  workspace: "MyApp.xcworkspace"
)
```
**xcclean**
```ruby
xcclean
```
**xctest**
```ruby
xcarchive(
  destination: "name=iPhone 5s,OS=8.1",
  scheme: "MyApp",
  workspace: "MyApp.xcworkspace"
)
```

**Comment by KrauseFx** [comment]
Thank you so much @dtrenz for working on this. Great result! :rocket:

**Pull request merged**